### DISCUSSION TITLE
How to use a shape from three-to-cannon?

### DISCUSSION THREAD

**Discussion by ffdead** [pmndrs/use-cannon#379] [Q&A] [answered]
I'm looking into using `three-to-cannon` for adding collisions to a GLB model that is unknown at compile time. 

But couldn't figure out how to use it with `use-cannon` so far.. is there a way to use an existing Shape?

Related discussion #182

**Comment by isaac-mason** [comment, marked as answer]
You can use the properties of an existing `cannon-es` shape as inputs to `use-cannon` shape hooks.

This is a bit messy, but this codesandbox demonstrates the approach: https://codesandbox.io/s/three-to-cannon-q0p282

**Comment by ffdead** [comment]
Oh wow, thank you so much @isaac-mason 🙏

**Reply by isaac-mason** [reply]
No worries! :)

**Reply by ffdead** [reply]
Extracted the solution from the codesandbox for anyone looking at this from the future 👋 

```
const shape = threeToCannon(ref.current, { type: ShapeType.HULL }).shape as ConvexPolyhedron
```

```
useConvexPolyhedron(
      () => ({
        args={[shape.vertices.map((v) => [v.x, v.y, v.z]), shape.faces, shape.faceNormals.map((f) => [f.x, f.y, f.z])]}
        ...
      })
)
```
